test/basicfuns.jl

```diff
@@ -135,7 +135,7 @@ end
     end
 
     @test isnan(logsubexp(Inf, Inf))
-    @test isnan(logsubexp(-Inf, -Inf))
+    @test logsubexp(-Inf, -Inf) ≡ -Inf
```

Comment:
Maybe test all combinations of `Inf` and `-Inf`? Also, why isn't `NaN` as second argument tested?

Reply:
I added more tests.